tests/pip_hub_repository/render_pkg_aliases/render_pkg_aliases_test.bzl

```diff
@@ -0,0 +1,251 @@
+
+_tests = []
+
+def _test_legacy_aliases(env):
+    actual = render_pkg_aliases(
+        bzl_packages = ["foo"],
+        repo_name = "pypi",
+    )
+
+    want = {
+        "foo/BUILD.bazel": """\
+package(default_visibility = ["//visibility:public"])
+
+alias(
+    name = "foo",
+    actual = ":pkg",
+)
+
+alias(
+    name = "pkg",
+    actual = "@pypi_foo//:pkg",
+)
+
+alias(
+    name = "whl",
+    actual = "@pypi_foo//:whl",
+)
+
+alias(
+    name = "data",
+    actual = "@pypi_foo//:data",
+)
+
+alias(
+    name = "dist_info",
+    actual = "@pypi_foo//:dist_info",
+)""",
+    }
+
+    env.expect.that_dict(actual).contains_exactly(want)
+
+_tests.append(_test_legacy_aliases)
+
+def _test_all_legacy_aliases_are_created(env):
+    actual = render_pkg_aliases(
+        bzl_packages = ["foo", "bar"],
+        repo_name = "pypi",
+    )
+
+    want_files = ["bar/BUILD.bazel", "foo/BUILD.bazel"]
+
+    env.expect.that_dict(actual).keys().contains_exactly(want_files)
+
+_tests.append(_test_all_legacy_aliases_are_created)
+
+def _test_bzlmod_aliases(env):
+    actual = render_pkg_aliases(
+        default_version = "3.2.3",
+        repo_name = "pypi",
+        rules_python = "rules_python",
+        whl_map = {
+            "bar-baz": ["3.2.3"],
+        },
+    )
+
+    want = {
+        "bar_baz/BUILD.bazel": """\
+package(default_visibility = ["//visibility:public"])
+
+alias(
+    name = "bar_baz",
+    actual = ":pkg",
+)
+
+alias(
+    name = "pkg",
+    actual = select(
+        {
+            "@@rules_python//python/config_settings:is_python_3.2.3": "@pypi_32_bar_baz//:pkg",
+            "//conditions:default": "@pypi_32_bar_baz//:pkg",
+        },
+    ),
+)
+
+alias(
+    name = "whl",
+    actual = select(
+        {
+            "@@rules_python//python/config_settings:is_python_3.2.3": "@pypi_32_bar_baz//:whl",
+            "//conditions:default": "@pypi_32_bar_baz//:whl",
+        },
+    ),
+)
+
+alias(
+    name = "data",
+    actual = select(
+        {
+            "@@rules_python//python/config_settings:is_python_3.2.3": "@pypi_32_bar_baz//:data",
+            "//conditions:default": "@pypi_32_bar_baz//:data",
+        },
+    ),
+)
+
+alias(
+    name = "dist_info",
+    actual = select(
+        {
+            "@@rules_python//python/config_settings:is_python_3.2.3": "@pypi_32_bar_baz//:dist_info",
+            "//conditions:default": "@pypi_32_bar_baz//:dist_info",
+        },
+    ),
+)""",
+    }
+
+    env.expect.that_dict(actual).contains_exactly(want)
+
+_tests.append(_test_bzlmod_aliases)
+
+def _test_bzlmod_aliases_with_no_default_version(env):
+    actual = render_pkg_aliases(
+        default_version = None,
+        repo_name = "pypi",
+        rules_python = "rules_python",
+        whl_map = {
+            "bar-baz": ["3.2.3", "3.1.3"],
+        },
+    )
+
+    want_key = "bar_baz/BUILD.bazel"
+    want_content = """\
+package(default_visibility = ["//visibility:public"])
+
+_NO_MATCH_ERROR = \"\"\"\\
```

Comment:
Ah, I guess I better get on https://github.com/bazelbuild/rules_testing/issues/65 so you can do e.g. `that_str(actual).matches("_NO_MATCH_ERROR*<everything else>")`. The long error message is nice, but embedding it in the test is brittle and distracting.

(I'm not asking to change this, it's fine as is; it's easy to update if we change the error message)